extensions/ql-vscode/src/query-history.ts

```diff
@@ -956,11 +956,11 @@ export class QueryHistoryManager extends DisposableObject {
       void this.tryOpenExternalFile(query.csvPath);
       return;
     }
-    await query.exportCsvResults(this.qs, query.csvPath, () => {
+    if (await query.exportCsvResults(this.qs, query.csvPath)) {
       void this.tryOpenExternalFile(
         query.csvPath
       );
-    });
+    }
```

Comment:
Should we be displaying an error in the `else` case here? I see we `reject()` where we were previously using `showAndLogErrorMessage()` but as far as I can tell we are then swallowing that error.

Reply:
@aeisenberg I don't think you've addressed this comment. It's less blocking the other others, so I'm happy to approve without any changes here if you don't think any changes are needed, but just checking it wasn't missed accidentally.

Reply:
Oh...sorry. I forgot to comment here. The error is not being swallowed. Take a look at `commandRunnerWithProgress`, you can see that there is an error barrier there that catches and logs all thrown and rejected errors. So, you don't ever need to explicitly catch and log an error when you're inside a command as long as that command is wrapped in `commandRunnerWithProgress`.

Reply:
Gotcha, thanks for the explanation 🙂